### PR TITLE
Fix wordrank max_iter_dump calculation. Fix #1216

### LINKED ISSUE #1216: Wrong calculation for max_iter_dump
In
https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/models/wrappers/wordrank.py#L144
Shouldnt this line `max_iter_dump = iter / dump_period * dump_period - 1` just be 
`max_iter_dump = iter - dump_period` ?

To reproduce try these parameters:
`model = Wordrank.train(wr_path, data, out_dir, iter=100, dump_period=5)`
It will error out with -
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ajkale/anaconda2/envs/wordrank/lib/python2.7/site-packages/gensim/models/wrappers/wordrank.py", line 146, in train
    copyfile('model_word_%d.txt' % max_iter_dump, 'wordrank.words')
  File "/home/ajkale/anaconda2/envs/wordrank/lib/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: 'model_word_99.txt'
```

Mainly because `max_iter_dump = iter / dump_period * dump_period - 1`  calculates max_iter_dump=99 instead of 95.

### PR DESCRIPTION
**Pull request by ajkl**
fix for https://github.com/RaRe-Technologies/gensim/issues/1216

### CODE REVIEW

**Comment by tmylk** [comment]
Travis tests re-ran after smart_open update

**Comment by tmylk** [comment]
Please add warning as in https://github.com/RaRe-Technologies/gensim/issues/1216#issuecomment-287048117

**Comment by parulsethi** [comment]
thanks for the improvement, please make default iter=90 (now that +1 iter is taken care of inside the  function), so that user don't get a warning at default parameters

**Comment by parulsethi** [comment]
@tmylk LGTM

**Comment by tmylk** [comment]
Thanks for the fix. Glad wordrank is being useful.

File: gensim/models/wrappers/wordrank.py [hunk 1]
```diff
@@ -113,6 +113,11 @@ def train(cls, wr_path, corpus_file, out_path, size=100, window=15, symmetric=1,
         with smart_open(meta_file, 'wb') as f:
             meta_info = "{0} {1}\n{2} {3}\n{4} {5}".format(numwords, numwords, numlines, cooccurrence_shuf_file, numwords, vocab_file)
             f.write(meta_info.encode('utf-8'))
+            
+        if iter % dump_period == 0:
+            iter += 1
+        else:
+            logger.warning('Resultant embedding would be from %d iteration', iter - iter % dump_period)
```
Comment (piskvorky) [inline gensim/models/wrappers/wordrank.py]: Seeing this warning tells the user nothing -- can you rephrase/add context/suggestions what to do about it?

Also, `iteration` => `iterations` and `dump_period` as an argument to `warning()`, rather than formatted string.

File: gensim/models/wrappers/wordrank.py [hunk 2]
```diff
@@ -141,7 +146,7 @@ def train(cls, wr_path, corpus_file, out_path, size=100, window=15, symmetric=1,
         output = utils.check_output(args=cmd)
 
         # use embeddings from max. iteration's dump
-        max_iter_dump = (iter - 1) - (iter - 1) % dump_period
+        max_iter_dump = iter - iter % dump_period
```
Comment (piskvorky) [inline gensim/models/wrappers/wordrank.py]: Use explicit brackets, don't rely on operator precedence.
